mne/time_frequency/tfr.py

```diff
@@ -1137,156 +1434,151 @@ def save(self, fname, overwrite=False):
-        The condition to load. If None, all conditions will be returned.
-        Defaults to None.
-
-    See Also
-    --------
-    write_tfrs
+    all_tfr : list of AverageTFR
+        The tfr datasets.
+    weights : list of float | str
+        The weights to apply to the data of each AverageTFR instance.
+        Can also be ``'nave'`` to weight according to tfr.nave,
+        or ``'equal'`` to use equal weighting (each weighted as ``1/N``).
 
     Returns
     -------
-    tfrs : list of instances of AverageTFR | instance of AverageTFR
-        Depending on `condition` either the TFR object or a list of multiple
-        TFR objects.
+    tfr : AverageTFR
+        The new TFR data.
 
     Notes
     -----
-    .. versionadded:: 0.9.0
+    .. versionadded:: 0.11.0
     """
+    tfr = all_tfr[0].copy()
+    if isinstance(weights, string_types):
+        if weights not in ('nave', 'equal'):
+            raise ValueError('Weights must be a list of float, or "nave" or '
+                             '"equal"')
+        if weights == 'nave':
+            weights = np.array([e.nave for e in all_tfr], float)
+            weights /= weights.sum()
+        else:  # == 'equal'
+            weights = [1. / len(all_tfr)] * len(all_tfr)
+    weights = np.array(weights, float)
+    if weights.ndim != 1 or weights.size != len(all_tfr):
+        raise ValueError('Weights must be the same size as all_tfr')
 
-    check_fname(fname, 'tfr', ('-tfr.h5',))
-
-    logger.info('Reading %s ...' % fname)
-    tfr_data = read_hdf5(fname, title='mnepython')
-    for k, tfr in tfr_data:
-        tfr['info'] = Info(tfr['info'])
+    ch_names = tfr.ch_names
+    for t_ in all_tfr[1:]:
+        assert t_.ch_names == ch_names, ValueError("%s and %s do not contain "
+                                                   "the same channels"
+                                                   % (tfr, t_))
+        assert np.max(np.abs(t_.times - tfr.times)) < 1e-7, \
+            ValueError("%s and %s do not contain the same time instants"
+                       % (tfr, t_))
 
-    if condition is not None:
-        tfr_dict = dict(tfr_data)
-        if condition not in tfr_dict:
-            keys = ['%s' % k for k in tfr_dict]
-            raise ValueError('Cannot find condition ("{0}") in this file. '
-                             'I can give you "{1}""'
-                             .format(condition, " or ".join(keys)))
-        out = AverageTFR(**tfr_dict[condition])
-    else:
-        out = [AverageTFR(**d) for d in list(zip(*tfr_data))[1]]
-    return out
+    # use union of bad channels
+    bads = list(set(tfr.info['bads']).union(*(t_.info['bads']
+                                              for t_ in all_tfr[1:])))
+    tfr.info['bads'] = bads
 
+    tfr.data = sum(w * t_.data for w, t_ in zip(weights, all_tfr))
+    tfr.nave = max(int(1. / sum(w ** 2 / e.nave
+                                for w, e in zip(weights, all_tfr))), 1)
```

Comment:
This is painfully not DRY with `combine_evoked`, you don't necessarily need to fix it here but please add a code comment to fix it later